### PR TITLE
Bug/1369 course grade override

### PR DESCRIPTION
**Pull request by steveswinsburg**
Delivers #1369 

- Adds the ability to set a course grade override and view the log. 
- Adds validation so the course grade entered is one of those in the selected grading schema
- Allows instructor to reset course grade and this is reflected in the log as the override being removed.
- Adds backend service to allow updating of course grades.
- Added a GbFeedbackPanel component as it is common to a bunch of pages. Refactored those pages to suit.

### CODE REVIEW

File: edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java [hunk 1]
```diff
@@ -3005,19 +3007,27 @@ private Double calculateCategoryScore(String studentUuid, Long categoryId, List<
 			}
 			
 			CourseGradeRecord gradeRecord = gradeRecords.get(0);
-
+			
+			System.out.println(ToStringBuilder.reflectionToString(gradeRecord, ToStringStyle.MULTI_LINE_STYLE));
```
Comment (payten) [inline edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java]: Debugging?

**Comment by steveswinsburg** [comment]
Thanks @payten, fixed that, however discovered a bug so don't merge this yet. Will push an update soon.

**Comment by steveswinsburg** [comment]
OK this is good to go now @payten :white_check_mark: 

*Changes made:*
* Backend service now more robust when there is no course grade record for a student - it fetches the data and adds a record.
* Dynamic course grade updating was not respecting the course grade override, it does now.

File: gradebookng/tool/src/java/org/sakaiproject/gradebookng/GradebookNgApplication.properties [hunk 2]
```diff
@@ -76,7 +77,12 @@ heading.updateungradeditems = Set Score for Empty Cells
 heading.gradelog = Grade Log for {0} ({1})
 heading.editcomment = Comment for {0} ({1}) - {2}
 heading.studentpage = Grade Report for {0}
+<<<<<<< HEAD
+heading.coursegrade = Course Grade Override for {0} ({1})
+heading.coursegradelog = Course Grade Override Log for {0} ({1})
+=======
 heading.zeroungradeditems = Set Zero Score for Empty Cells
+>>>>>>> master
```
Comment (payten) [inline gradebookng/tool/src/java/org/sakaiproject/gradebookng/GradebookNgApplication.properties]: Eep, some merge remnants!

**Comment by steveswinsburg** [comment]
Fixed. Crap. I didnt see that on the master merge :(

**Pull request merged**